chapters/lexicalstructure.tex

```diff
@@ -158,9 +158,9 @@ \subsection{Floating Point Numbers}\doublelabel{floating-point-numbers}
 \subsection{Integer Literals}\doublelabel{integer-literals}
 
 Literals of type \lstinline!Integer! are sequences of decimal digits, e.g. as in the
-integer numbers \lstinline!33, 0, 100, 30030044!. {[}\emph{Negative numbers are
+integer numbers \lstinline!33!, \lstinline!0!, \lstinline!100!, \lstinline!30030044!. {[}\emph{Negative numbers are
```

Comment:
Going back to the original question of #2122: Shouldn't section 2.4.1 _Floating Point Numbers_ and 2.4.2 _Integer Literals_ refer to the lexer rules we've now been trying so hard to get right?

Reply:
Yes, that would make sense. 

> 
> 
> Going back to the original question of #2122: Shouldn't section 2.4.1 _Floating Point Numbers_ and 2.4.2 _Integer Literals_ refer to the lexer rules we've now been trying so hard to get right?
> 
> Now that the lexer rule for `Real` literals has become quite complicated, it seems like a bad idea to also redundantly say the same thing using natural language. A couple of examples is probably all we need in the text.

Yes, that would be a good idea.

Reply:
On second thought just removing the entire natural language text seems a bit excessive. The part that is difficult to explain in natural language is what is optional. I would propose something like the following text:

> A floating point number is expressed as a decimal number in the form of
> a sequence of decimal digits followed by a decimal point, followed by decimal digits,
> followed by an exponent, indicated by E or e followed by a sign 
> and one or more decimal digits. The various parts can be omitted, see the grammar for 
> details and also the examples below. 
(Or possibly dropping "decimal" except for the first one.)

Reply:
This sounds very good to me.

Reply:
As of the latest commit and the comments to it, I think we're done with this conversation.  Resolving.